### ISSUE TITLE
Frequently crashes on flutter 2.8

### ISSUE THREAD

**Issue by manjav** [Abedalkareem/games_services#50]
![image](https://user-images.githubusercontent.com/3495382/146922031-84b66a20-046f-41a9-ae83-71d768e4ac00.png)

**Google Console:**
```java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5533)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5574)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2309)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8587)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
Caused by: java.lang.NullPointerException: 
  at com.abedalkareem.games_services.GamesServicesPlugin.finishPendingOperationWithError (GamesServicesPlugin.java:2)
  at com.abedalkareem.games_services.GamesServicesPlugin.onActivityResult (GamesServicesPlugin.java:84)
  at io.flutter.embedding.engine.FlutterEngineConnectionRegistry$FlutterEngineActivityPluginBinding.onActivityResult (FlutterEngineConnectionRegistry.java:25)
  at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.onActivityResult (FlutterEngineConnectionRegistry.java:20)
  at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult (FlutterActivityAndFragmentDelegate.java:51)
  at io.flutter.embedding.android.FlutterActivity.onActivityResult (FlutterActivity.java:10)
  at android.app.Activity.dispatchActivityResult (Activity.java:8550)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5526)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5574)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2309)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8587)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
```

**Game Analytics:**
```- Type of exception: java.lang.RuntimeException
- Exception message: Failure delivering result ResultInfo{who=null, request=9000, result=-1, data=Intent { (has extras) }} to activity {game.block.puzzle.drop.the.number.merge/game.block.puzzle.drop.the.number.merge.MainActivity}: java.lang.NullPointerException
- Thread name: main
- Stacktrace: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=9000, result=-1, data=Intent { (has extras) }} to activity {game.block.puzzle.drop.the.number.merge/game.block.puzzle.drop.the.number.merge.MainActivity}: java.lang.NullPointerException
	at android.app.ActivityThread.deliverResults(ActivityThread.java:4427)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4471)
	at android.app.ActivityThread.-wrap20(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1796)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6666)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:583)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
Caused by: java.lang.NullPointerException
	at d.b.a.a.i(GamesServicesPlugin.kt:1)
	at d.b.a.a.j(GamesServicesPlugin.kt:7)
	at d.b.a.a.a(GamesServicesPlugin.kt:4)
	at io.flutter.embedding.engine.d$c.g(FlutterEngineConnectionRegistry.java:2)
	at io.flutter.embedding.engine.d.a(FlutterEngineConnectionRegistry.java:4)
	at io.flutter.embedding.android.f.n(FlutterActivityAndFragmentDelegate.java:4)
	at io.flutter.embedding.android.e.onActivityResult(FlutterActivity.java:2)
	at android.app.Activity.dispatchActivityResult(Activity.java:7296)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:4423)
	... 9 more
```

**Comment by manjav** [comment]
![image](https://user-images.githubusercontent.com/3495382/146927932-8dc2a4c3-a036-4ca7-83c8-826310fa5c98.png)

I updated flutter to 2.8 and no other changes on 3 December

**Comment by manjav** [comment]
`pendingOperation` is null but the null-check of kotlin went to crash

https://github.com/Abedalkareem/games_services/blob/master/games_services/android/src/main/kotlin/com/abedalkareem/games_services/GamesServicesPlugin.kt#:~:text=Log.i(-,pendingOperation,-!!.method%2C%20%22error%22)

**Comment by Abedalkareem** [comment]
Thank you for reporting, working on that.

**Comment by manjav** [comment]
I updated kotlin version in gradle and almost fix crashes

**Comment by Abedalkareem** [comment]
The issue fixed in version 2.0.2.

Thank you.
